tardis/plugins/sqliteregistry.py

```diff
@@ -49,10 +50,15 @@ async def async_execute(self, sql_query: str, bind_parameters: dict):
             self.thread_pool_executor, self.execute, sql_query, bind_parameters
         )
 
+    @contextmanager
     def connect(self):
-        return sqlite3.connect(
+        con = sqlite3.connect(
             self._db_file, detect_types=sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES
         )
+        try:
+            yield con
+        finally:
+            con.close()
```

Comment:
I think it is better if the context manager also automatically ``commit``s, which allows the other code to use just one ``with``. This should keep the code simpler and avoid errors; since ``commit`` is idempotent, it's still possible to use smaller transactions (via ``with``/``commit``) and one can ``rollback`` to discard a transaction.
```suggestion
        con = sqlite3.connect(
            self._db_file, detect_types=sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES
        )
        try:
            yield con
        finally:
            con.commit()
            con.close()
```

Reply:
That is unfortunately not all. We need to rollback, if an Exception is thrown. Not sure what exception can occur, but we would need to figure that out to avoid a catchall one.

I never tried that, but what do think about 

```
con = sqlite3.connect(
            self._db_file, detect_types=sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES
        )
        try:
            with con:
                yield con
        finally:
            con.close()
```
?

Reply:
Ah, yes, you are correct – I mistakenly assumed that ``with connection:`` would always commit. Your suggested code should work.

Reply:
Done!